samples/hello_cloudevents/src/multiple_google_events.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * The samples in the file demonstrate using the `isGoogleEvent` type predicate
+ * provided by the @google/events library to implement a function that accepts
+ * multiple different Google CloudEvents.
+ */
+import * as functions from '@google-cloud/functions-framework';
+import {isGoogleEvent} from '../google_events';
+
+functions.cloudEvent('MultiGoogleEvent1', ce => {
+  if (!isGoogleEvent(ce)) {
+    throw "woops this shouldn't happen!";
+  }
+  // in this branch the typechecker knows ce is a GoogleCloudEvent
+  switch (ce.type) {
```

Comment:
I am a little bit concerned about how this function paradigm would scale to less loosely structured languages like Golang.

I think ideally we should strive for declarative type functionality that has the best experience for a single type. Currently, GCF samples only handle single types as well.